RFCs/FS-1331-Allow opens in type and expression scopes.md

```diff
@@ -0,0 +1,99 @@
+# F# RFC FS-1331 - Allow opens in type and expression scopes
+
+The design suggestion [Allow the use of open at type and expression scopes](https://github.com/fsharp/fslang-suggestions/issues/96) has been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/96)
+- [x] Approved in principle
+- [x] [Implementation](https://github.com/dotnet/fsharp/pull/18814)
+- [x] [Discussion](https://github.com/fsharp/fslang-design/discussions/812)
+
+# Summary
+
+This RFC proposes to allow the use of `open` at type and expression scopes. Type/expression-scoped `open`s will work as same as module-scoped `open`, that is, modules or types with `[<RequireQualifiedAccess>]` cannot be opened.
+
+# Motivation
+
+By allowing this, we can 
+1. Picking a certain set of operator overloads in some tight context like `Checked`.
+2. Making the `open`s span less than the whole module/namespace.
+
+# Detailed design
+
+1. Expression-scoped `open` is an expression that opens a module in the body expression scope, and its type is the body's type.
+
+```fsharp
+((open System
+  Int32.MaxValue + 1   // The body expression
+): int)
+
+let test () =
+    open global.System
+    printfn "%d" (Int32.MaxValue + 1)
+
+    open type System.Int32
+    open Checked
+    printfn "%d" (MaxValue + 1)
+```
+
+2. Type-scoped `open` is a statement that opens a module in the type's following scope. It can be used any type definitions, type expressions and the `with` section of a record/union/exception type.
+
+
+```fsharp
+type C() =
+    do printfn "%d" Int32.MaxValue   // <- Cannot find the `Int32` here
+    open System
+    do printfn "%d" Int32.MaxValue
+    member _.M() = open type Int32; MaxValue
+
+
+type System.Int32 with
+    open type System.Math
+    member this.Abs111 = Abs(this)
+
+
+type A = A of int
+    with
+        open System
+// ....
+```
+
+# Drawbacks
+
+TODO
```

Comment:
Are there any foreseen deviations from a module level `open` ? I assume those can arise from the implementation.
If yes, I would like to have them listed here (inconsistency being the drawback).